presto-main/src/main/java/com/facebook/presto/execution/scheduler/ConsistentHashingNodeProvider.java

```diff
@@ -40,15 +41,15 @@
     private final NavigableMap<Integer, InternalNode> candidates;
     private final int nodeCount;
 
-    static ConsistentHashingNodeProvider create(Collection<InternalNode> nodes, int weight)
+    static Optional<ConsistentHashingNodeProvider> create(Collection<InternalNode> nodes, int weight)
```

Comment:
`ConsistentHashingNodeProvider.create(..)` will always create an instance. So lets not return Optional. Lets leave it as is ..so that it return `ConsistentHashingNodeProvider`. 
Instead lets change the caller.